src/Sarif.Multitool.Library/RewriteCommand.cs

```diff
@@ -64,6 +65,25 @@ public int Run(RewriteOptions options)
                     reformattedLog = new SortingVisitor().VisitSarifLog(reformattedLog);
                 }
 
+                if (options.NormalizeForGitHub)
+                {
+                    if ((reformattedLog.Runs?.Any(r => r.Artifacts?.Any(a => a.Location?.Uri?.IsAbsoluteUri == true) == true) == true) ||
```

Comment:
>reformattedLog [](http://example.com/codeflow?start=25&length=14)

So I don't like these two lines of code which will traverse the log file *twice*.

Can't we simply invoke the RebaseUriVisitor in cases when those args are non-null.

And then make the GitHubIngestionVisitor throw new arg exception on encountering the any absolute uri as you detect here.

Reply:
made the suggested change.
Though we can not simply throw in GitHubIngestionVisitor, because it is a same method `VisitArtifactLocation` to visit both of it in result and also the originalUriBaseIds,
we allow one and not the other to be absolute.
Currently I does not check and throw, let me know your instruction.